src/modules/enrollment/components/dashboardPages/EnrollmentCeEventsPage.tsx

```diff
@@ -62,21 +62,27 @@ const EnrollmentCeEventsPage = () => {
 
   const canEditCeEvents = enrollment?.access?.canEditEnrollments || false;
 
-  const { onSelectRecord, viewRecordDialog, editRecordDialog, openFormDialog } =
-    useViewEditRecordDialogs({
-      variant: canEditCeEvents ? 'view_and_edit' : 'view_only',
-      inputVariables: { enrollmentId },
-      formRole: RecordFormRole.CeEvent,
-      recordName: 'CE Event',
-      evictCache,
-      deleteRecordDocument: DeleteCeEventDocument,
-      deleteRecordIdPath: 'deleteCeEvent.ceEvent.id',
-      maxWidth: 'sm',
-      localConstants,
-      projectId: enrollment?.project.id,
-    });
+  const {
+    onSelectRecord,
+    viewRecordDialog,
+    editRecordDialog,
+    openFormDialog,
+    formNotFound,
+  } = useViewEditRecordDialogs({
```

Comment:
I wouldn't think this is specific to CE Events, is it? There are plenty of other pages that use this pattern: I'm seeing CE Assessments, CLS, Case Notes, HMIS Participation, and CE Participation.

This approach feels a little tricky to maintain, because each component that uses the `useViewEditRecordDialogs` hook needs to realize that it's possible for the hook to fail, and that the component should look at `formNotFound` and render a NotFound component. The fact that the form isn't found feels like something that shouldn't be the concern of the dialog hook. IMO would make more sense if the dialog hook just throw an error if the form wasn't found–– because if the form wasn't found, it cannot render a dialog safely.

I'd propose that we use the same logic that we already use to determine whether the given page shows up in the left nav. For CE Events, that's:
https://github.com/greenriver/hmis-frontend/blob/84fccbcbedc112da322fe8f855f0ea379dccd73a/src/modules/enrollment/hooks/useEnrollmentDashboardNavItems.tsx#L50-L52

I think it would be ideal if the app could say "you're trying to navigate to `/events` but events wasn't included in the nav items, so I'm going to return NotFound [or redirect you to the enrollment dash overview]" without bothering to attempt to load the page itself. If we can do that in a way that is extensible to all dashboard pages, we don't need to worry about it in each pages render logic. (And we can raise an error if the form isn't found, because we don't expect that)


We do already have the enabledFeatures list in the context:
```
const { enabledFeatures } = useEnrollmentDashboardContext();
  if (!enabledFeatures.includes(DataCollectionFeatureRole.CeEvent)) {
    return <NotFound />;
  }
```

Looking at the `protected.tsx` routes, I wonder if we can just introduce a prop to `EnrollmentRoute` to this effect:
```
            element: (
              <EnrollmentRoute
                dataCollectionFeature={DataCollectionFeatureRole.CeEvent|
                redirectRoute={EnrollmentDashboardRoutes.ENROLLMENT_OVERVIEW}
              >
                <EnrollmentCeEventsPage />
              </EnrollmentRoute>
            ),
```
Similar to how `EnrollmentRoute` handles returning NotFound (or redirect) if the user lacks permission, it could likewise return NotFound (or redirect) if the Data Collection Feature is not listed in the enrollment dashboard contexts' `enabledFeatures` list.

Reply:
Thanks @gigxz  I like that idea, it's really elegant. 

Do you think we should implement something similar in `ProjectRoute`, for HMIS Participation, CE Participation, Current living situation, and maybe external form submissions?

Reply:
Yes I think it would make sense to have in the project routes as well, since it uses the same concept of data collection features (For Services and Current Living Situations). I don't think the same would apply to HMIS Participation/ CE participation since those aren't "optional" features.